File: Demo/DemoWidget/TodayViewController.m
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 */
@import MobileCenter;
@import MobileCenterAnalytics;
@import MobileCenterCrashes;
#import "TodayViewController.h"
@import NotificationCenter;

@interface TodayViewController () <NCWidgetProviding>

@end

static BOOL MobileCenterSDKIsSarted = NO;

@implementation TodayViewController

- (void)viewDidLoad {
  [super viewDidLoad];

  if (!MobileCenterSDKIsSarted) {
    [MSMobileCenter setLogLevel:MSLogLevelVerbose];
    [MSMobileCenter start:@"7dfb022a-17b5-4d4a-9c75-12bc3ef5e6b7"
             withServices:@[ [MSAnalytics class], [MSCrashes class] ]];
    MobileCenterSDKIsSarted = YES;
  }
  // Do any additional setup after loading the view from its nib.
}

- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

- (void)widgetPerformUpdateWithCompletionHandler:(void (^)(NCUpdateResult))completionHandler {
  // Perform any setup necessary in order to update the view.

  // If an error is encountered, use NCUpdateResultFailed
  // If there's no update required, use NCUpdateResultNoData
  // If there's an update, use NCUpdateResultNewData

  completionHandler(NCUpdateResultNewData);
}

- (IBAction)trackEventClick:(id)sender {
  [MSAnalytics trackEvent:@"test"];
}

- (IBAction)crashClick:(id)sender {

  /* Trigger a crash */
  NSArray *array = [NSArray array];
  [array objectAtIndex:23];
}

@end

File: SasquatchMac/SasquatchMacSwift-Preview Today Extension/TodayViewController.swift

// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import Cocoa
import NotificationCenter
import AppCenter
import AppCenterCrashes
class TodayViewController: NSViewController, NCWidgetProviding {
    @IBOutlet weak var extensionLabel: NSTextField!
    
    override var nibName: NSNib.Name? {
        return NSNib.Name("TodayViewController")
    }
    var didStartAppCenter = false;
    override func viewDidLoad() {
        
        super.viewDidLoad()
        extensionLabel.stringValue = "Run #\(UUID().uuidString)"
        if (!didStartAppCenter){
            MSAppCenter.setLogLevel(.verbose);
            MSAppCenter.start("3ccfe7f5-ec01-4de5-883c-f563bbbe147a", withServices: [MSCrashes.self])
            didStartAppCenter = true;
        }
        // Do any additional setup after loading the view from its nib.
    }
    @IBAction func crashMe(_ sender: Any) {
        let buf: UnsafeMutablePointer<UInt>? = nil;
        buf![1] = 1;
    }
    func widgetPerformUpdate(completionHandler: (@escaping (NCUpdateResult) -> Void)) {
        // Update your data and prepare for a snapshot. Call completion handler when you are done
        // with NoData if nothing has changed or NewData if there is new data since the last
        // time we called you
        completionHandler(.noData)
    }
}
